tests/pathtourl.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include <cstdlib>
+#include <iostream>
+
+#include "ada.h"
+
+
+/**
+ * Node.js has a function called url.pathToFileURL(path).
+ * https://nodejs.org/api/url.html#urlpathtofileurlpath
+ *
+ * It is explained as follows...
+ *
+ * This function ensures that path is resolved absolutely,
+ * and that the URL control characters are correctly encoded when converting into a File URL.
+ *
+ * new URL('/foo#1', 'file:');           // Incorrect: file:///foo#1
```

Comment:
On Safari both this example and `new URL("file:///foo#1")` has a href value of `file:///foo#1`

Reply:
Whereas in `let u = new URL('file:///test'); u.pathname = "/foo#1";` `foo#1` is considered as query and percent encoded.

Reply:
@anonrig Right. So if you have a path that is "/foo#1", then it needs to be percent encoded. You can either do it yourself, or use ada:

```C++
     ada::result url = ada::parse("file://");
     if(!url->set_pathname("/foo#1")) { return false; }
     if(url->get_href() != "file:///foo%231") { return false; }
```

So there is no need for a special function (e.g., pathToFileURL) in this instance. Just use the path setter from ada.

Correct?

Reply:
Path setter would do it. I also assume that path setter changes `%` to `%25` right? Since it's the other requirement of the function. If yes, we don't have to do a for loop in here: https://github.com/nodejs/node/blob/main/src/node_url.cc#L332